fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/ChirpHandler.java

```diff
@@ -40,4 +45,33 @@ public static void handleChirpAdd(HandlerContext context, AddChirp addChirp) {
 
     lao.updateChirp(messageId, chirp);
   }
+
+  /**
+   * process a DeleteChirp message.
+   *
+   * @param context the HandlerContext of the message
+   * @param deleteChirp the data of the message that was received
+   */
+  public static void handleDeleteChirp(HandlerContext context, DeleteChirp deleteChirp) {
+    LAORepository laoRepository = context.getLaoRepository();
+    String channel = context.getChannel();
+
+    Lao lao = laoRepository.getLaoByChannel(channel);
+
+    Optional<Chirp> chirpOptional = lao.getChirp(deleteChirp.getChirpId());
+    Chirp chirp;
+
+    if (!chirpOptional.isPresent()) {
+      throw new IllegalArgumentException("The chirp does not exist");
```

Comment:
There exist some exception that are more specific to this case, For example DataHandlingException, InvalidDataException, etc, they can be found in the package com.github.dedis.popstellar.utility.error